File: fe2-android/app/src/test/java/com/github/dedis/popstellar/model/network/method/QueryTest.java
package com.github.dedis.popstellar.model.network.method;

import com.github.dedis.popstellar.model.objects.Channel;

import junit.framework.TestCase;

import java.util.Objects;

import static org.junit.Assert.assertNotEquals;

public class QueryTest extends TestCase {
  private static final Channel CHANNEL = Channel.fromString("root/stuff");
  private static final int ID = 12;
  private static final Query QUERY =
      new Query(CHANNEL, ID) {
        @Override
        public String getMethod() {
          return "foo";
        }
      };

  public void testTestEquals() {
    Query query2 =
        new Query(Channel.fromString("nonRoot/stuff"), 12) {
          @Override
          public String getMethod() {
            return "foo";
          }
        };
    Query query3 =
        new Query(CHANNEL, 11) {
          @Override
          public String getMethod() {
            return "foo";
          }
        };
    assertEquals(QUERY, QUERY);
    assertNotEquals(null, QUERY);
    assertNotEquals(query2, QUERY);
    assertNotEquals(query3, QUERY);
  }

  public void testTestHashCode() {
    assertEquals(Objects.hash(Objects.hash(CHANNEL), ID), QUERY.hashCode());
  }
}

File: fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/model/network/method/QueryTest.kt
package com.github.dedis.popstellar.model.network.method

import com.github.dedis.popstellar.model.objects.Channel.Companion.fromString
import junit.framework.TestCase
import org.junit.Assert
import java.util.Objects

class QueryTest : TestCase() {
  fun testTestEquals() {
    val query2: Query = object : Query(fromString("nonRoot/stuff"), 12) {
      override val method: String
        get() = "foo"
    }
    val query3: Query = object : Query(CHANNEL, 11) {
      override val method: String
        get() = "foo"
    }
    assertEquals(QUERY, QUERY)
    Assert.assertNotEquals(null, QUERY)
    Assert.assertNotEquals(query2, QUERY)
    Assert.assertNotEquals(query3, QUERY)
  }

  fun testTestHashCode() {
    assertEquals(Objects.hash(Objects.hash(CHANNEL), ID), QUERY.hashCode())
  }

  companion object {
    private val CHANNEL = fromString("root/stuff")
    private const val ID = 12
    private val QUERY: Query = object : Query(CHANNEL, ID) {
      override val method: String
        get() = "foo"
    }
  }
}